src/Goutte/Client.php

```diff
@@ -11,33 +11,10 @@
 namespace Behat\Mink\Driver\Goutte;
 
 use Goutte\Client as BaseClient;
-use Symfony\Component\BrowserKit\Response;
 
 /**
  * Client overrides to support Mink functionality.
  */
 class Client extends BaseClient
 {
-    /**
-     * Reads response meta tags to guess content-type charset.
-     *
-     * @param Response $response
-     *
-     * @return Response
-     */
-    protected function filterResponse($response)
-    {
-        $contentType = $response->getHeader('Content-Type');
-
-        if (!$contentType || false === strpos($contentType, 'charset=')) {
-            if (preg_match('/\<meta[^\>]+charset *= *["\']?([a-zA-Z\-0-9]+)/i', $response->getContent(), $matches)) {
```

Comment:
This detection of the charset from the HTML is supported by symfony/dom-crawler since 2.2